gcc/common/config/riscv/riscv-common.c

```
@@ -513,6 +513,13 @@ riscv_subset_list::parse (const char *arch, location_t loc)
   if (p == NULL)
     goto fail;
 
+  if (*p != '\0')
+    {
+      error_at (loc, "-march=%s: unexpected ISA string at end: %<%s%>",
+               arch, p);
+      return NULL;
```

`goto fail;` instead of `return NULL;` here, because subset_list need to free